core/src/main/java/com/datastrato/gravitino/storage/relational/utils/POConverters.java

```diff
@@ -889,39 +885,64 @@ public static List<GroupRoleRelPO> initializeGroupRoleRelsPOWithVersion(
           .withNamespace(namespace)
           .withProperties(JsonUtils.anyFieldMapper().readValue(rolePO.getProperties(), Map.class))
-          .withSecurableObject(securableObject)
+          .withSecurableObjects(fromSecurablePersistObjects(rolePO.getSecurableObjects()))
           .withAuditInfo(
               JsonUtils.anyFieldMapper().readValue(rolePO.getAuditInfo(), AuditInfo.class))
           .build();
     } catch (JsonProcessingException e) {
       throw new RuntimeException("Failed to deserialize json object:", e);
     }
   }
+
+  private static List<SecurableObject> fromSecurablePersistObjects(String persisteSecurableObjects)
+      throws JsonProcessingException {
+    List<Map<String, Object>> securablePersistObjects =
+        JsonUtils.anyFieldMapper().readValue(persisteSecurableObjects, List.class);
+    return securablePersistObjects.stream()
+        .map(
+            persistObject -> {
+              List<Privilege> privileges = Lists.newArrayList();
+              List<Map<String, String>> privilegeMaps =
+                  (List<Map<String, String>>) persistObject.get(SECURABLE_OBJECT_PRIVILEGES);
+              for (Map<String, String> privilegeMap : privilegeMaps) {
+                if (Privilege.Condition.ALLOW
+                    .name()
+                    .equals(privilegeMap.get(PRIVILEGE_CONDITION))) {
+                  privileges.add(Privileges.allow(privilegeMap.get(PRIVILEGE_NAME)));
+                } else {
+                  privileges.add(Privileges.deny(privilegeMap.get(PRIVILEGE_CONDITION)));
+                }
+              }
+              return SecurableObjects.parse(
+                  (String) persistObject.get(SECURABLE_OBJECT_FULL_NAME),
+                  SecurableObject.Type.valueOf((String) persistObject.get(SECURABLE_OBJECT_TYPE)),
+                  privileges);
+            })
+        .collect(Collectors.toList());
+  }
+
+  private static List<Map<String, Object>> generateSecurablePersistObjects(
+      List<SecurableObject> securableObjects) {
+    List<Map<String, Object>> securablePersistObjects = Lists.newArrayList();
+    for (SecurableObject securableObject : securableObjects) {
+      Map<String, Object> securableObjectMap = Maps.newHashMap();
+      securableObjectMap.put(SECURABLE_OBJECT_FULL_NAME, securableObject.fullName());
+      securableObjectMap.put(SECURABLE_OBJECT_TYPE, securableObject.type().name());
+      List<Map<String, String>> privileges = Lists.newArrayList();
+      for (Privilege privilege : securableObject.privileges()) {
+        Map<String, String> privilegeMap = Maps.newHashMap();
+        privilegeMap.put(PRIVILEGE_NAME, privilege.name().name());
+        privilegeMap.put(PRIVILEGE_CONDITION, privilege.condition().name());
+        privileges.add(privilegeMap);
+      }
+      securableObjectMap.put(SECURABLE_OBJECT_PRIVILEGES, privileges);
+      securablePersistObjects.add(securableObjectMap);
+    }
+    return securablePersistObjects;
+  }
```

Comment:
I think you'd better add more tests about this part.

Reply:
I ever tried to use anyFieldMapper. But it can't serialize the fields privileges correctly. Because privilege doesn't have field.
For compatible issues, our way is similar to anyFiedlMapper. We can guarantee the compatibility.